From 94ae5db7d017480e103aa6f07cc86648b613d114 Mon Sep 17 00:00:00 2001
From: "Yann E. MORIN" <yann.morin.1998@free.fr>
Date: Sun, 18 Dec 2016 15:37:27 +0100
Subject: [PATCH] libxfs: do not try to run the crc32selftest

Even though the crc32selftest is natively compiled (because it is to be
executed), it fails in cross-compilation as the host may lack the
required headers, like uuid/uuid.h (e.g. in a minimal environment).

Moreover, running the crc32selftest natively is completely wrong,
because it passing on the host does not mean it would still pass n the
target (because endianness or bitness or alignment differences).

So, just disable running the crc32selftest altogether.

Note that there's a remaining bug-in-hiding, because the crc32 table
generator is natively built, but with the target CFLAGS.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
Signed-off-by: "Fabrice Fontaine" <fontaine.fabrice@gmail.com>
[Update for 4.18.0: crc32 has been moved from libxfs to libfrog]
[Julien: rebased on v6.4.0]
Signed-off-by: Julien Olivain <ju.o@free.fr>
Upstream: patch to be superseded by https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/commit/?id=e51b89e657af7843d8c7aea3df49fe1f0f1d4ac2
---
 libfrog/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libfrog/Makefile b/libfrog/Makefile
index f292afe3..1d0d2083 100644
--- a/libfrog/Makefile
+++ b/libfrog/Makefile
@@ -57,9 +57,9 @@ ifeq ($(HAVE_GETMNTENT),yes)
 LCFLAGS += -DHAVE_GETMNTENT
 endif
 
-LDIRT = gen_crc32table crc32table.h crc32selftest
+LDIRT = gen_crc32table crc32table.h
 
-default: crc32selftest ltdepend $(LTLIBRARY)
+default: ltdepend $(LTLIBRARY)
 
 crc32table.h: gen_crc32table.c crc32defs.h
 	@echo "    [CC]     gen_crc32table"
-- 
2.45.2

